core/kernel/panic.c

```diff
@@ -26,11 +26,14 @@
  */
 
 #include <kernel/panic.h>
+#include <kernel/thread.h>
 #include <trace.h>
 
 void __panic(const char *file __maybe_unused, int line __maybe_unused,
 		const char *func __maybe_unused)
 {
+	uint32_t __unused exceptions = thread_mask_exceptions(THREAD_EXCP_ALL);
+
 	EMSG_RAW("PANIC: %s %s:%d\n", func, file, line);
 	while (1)
 		;
```

Comment:
Yes, ideally we would send an SGI to all the other cores (as I suppose the Linux kernel does) to trap them. But that's a bit more complicated than what I'm doing here and this still improves a lot compared to before.